### PR TITLE
UWP: better mouse interaction support

### PR DESCRIPTION
**Pull request by paulrouget**

### CODE REVIEW

**Comment by jdm** [comment]
I filed #24530 after observing it in a desktop glutin window - are there similar changes we need to make to that port as well?

**Comment by paulrouget** [comment]
> I filed #24530 after observing it in a desktop glutin window - are there similar changes we need to make to that port as well?

Good to know. This is actually a different issue with the pointer event polyfil. Which I worked around in the latest bbjs demo update.

This PR doesn't fix this issue for desktop, but does fix the behavior for UWP builds.

**Comment by paulrouget** [comment]
The bbjs demos now use a better workaround for #24363. The double click issue (#24530) doesn't exist anymore.

But still need that fix for the non-glutin port.

File: support/hololens/ServoApp/ServoControl/ServoControl.cpp [hunk 1]
```diff
@@ -93,15 +96,50 @@ void ServoControl::OnSurfaceManipulationDelta(
...
+    }
+
+    if (!b.has_value() && mPressedMouseButton.has_value()) {
+      auto b = *mPressedMouseButton;
+      mPressedMouseButton = {};
+      RunOnGLThread([=] { mServo->MouseUp(x, y, b); });
+      e.Handled(true);
+    }
+
+    if (b.has_value()) {
```
Comment (jdm) [inline support/hololens/ServoApp/ServoControl/ServoControl.cpp]: Won't this also trigger when we've just run a MouseUp event in the previous branch?

**Review by jdm — CHANGES_REQUESTED** [review]

File: support/hololens/ServoApp/ServoControl/ServoControl.cpp (hunk 1 above)
Comment (paulrouget) [reply support/hololens/ServoApp/ServoControl/ServoControl.cpp]: It won't. I think the confusion comes from the very non descriptive variable name (`b`) which is used twice. I will change that.

**Pull request merged**